Extension improvemnts

ironnysh opened the pull request:
The main changes include:
- Added styles.
- Improved UX.
- Added documentation.
- Updated graphic assets.

ironnysh committed:
Extension improvemnts

The main changes include:
- Added styles.
- Improved UX.
- Added documentation.
- Updated graphic assets.

angelikatyborska commented on .gitignore:
```diff
@@ -1,2 +1,3 @@
 .amo-upload-uuid
 web-ext-artifacts/*.zip
+.DS_Store
```
I have a strong opinion that each developer is responsible for maintaining their own global `.gitignore` file based on their personal choices like OS and text editor, and a project's `.gitignore` should only contain files specific to the project. It wouldn't be maintainable for a single person to know all of the possible temporary and hidden files from all of the OSes and text editors...

```suggestion
```

ironnysh replied:
Interesting! I like that :-)
I'll update the PR

angelikatyborska commented on CHANGELOG.md:
```diff
@@ -1,4 +1,11 @@
-# Chanelog
+# Changelog
+
+## 1.3 (2025-05-06)
```
Note to self: change date before merging

angelikatyborska commented on CHANGELOG.md:
```diff
@@ -1,4 +1,11 @@
+- Add styles.
+- Improve UX.
+- Add documentation.
+- Update graphic assets.
```
I created a custom icon that is not the Storyblok logo on purpose. This is my private project, not approved by Storyblok, and I don't want to get into any legal trouble for using their logo. Can you please bring back the original icon?

ironnysh replied:
Sorry, hadn't thought about this

angelikatyborska commented on README.md:
```diff
@@ -1,13 +1,54 @@
+> [!IMPORTANT]
+> Safari removes temporary extensions after 24 hours or when you quit Safari.
+
+## Set up
```
Thank you so much for the extra documentation 🙏

angelikatyborska commented on style.css:
```diff
@@ -0,0 +1,198 @@
+thead,
+p,
+label {
+  font-stretch: 65%;
```
I find that the drastically changing font-stretch on all the various elements makes everything looks distorted and unreadable. Can you please remove all font-stretches?

angelikatyborska commented on options/index.html:
```diff
@@ -1,98 +1,75 @@
+            xmlns="http://www.w3.org/2000/svg"
+            aria-hidden="true"
+            title="Explore the code on GitHub"
+          >
```
I'm not really sure what this link was supposed to be or how it was supposed to look like. There is text in a paragraph that says "Open on Storyblok browser extension". Such text "open on..." would suggest this is already supposed to be a link, not a paragraph. Then there is an icon link, that leads to this Github repo that's titled "Explore the code on Github", and I'm not sure how this is related to "Open on Storyblok browser extension" 🤔 And it doesn't look right in Firefox when testing. The icon is gigantic.
<img width="698" alt="SCR-20250516-jzlo" src="https://github.com/user-attachments/assets/965634eb-a788-4edc-85b3-48008fa30751" />

Can we just remove all of this? I don't think the extension's settings page needs to link to this repo.

angelikatyborska commented on options/index.js:
```diff
@@ -9,6 +9,7 @@ function saveOptions(e) {
       name: formData.get(`spaces[${i}].name`),
       accessToken: formData.get(`spaces[${i}].accessToken`),
       rootStorySlug: formData.get(`spaces[${i}].rootStorySlug`),
+      spaceLegend: formData.get(`spaces[${i}].id`),
```
What's this?

ironnysh replied:
It displays the space ID in the `<legend>` :
<img width="463" alt="" src="https://github.com/user-attachments/assets/b4d79730-49c9-4813-b547-d929fc16cd9b" />

angelikatyborska commented on options/index.js:
```diff
@@ -25,52 +26,56 @@ function saveOptions(e) {
-    saveStatus.textContent = ''
-  }, 3000)
+    saveStatus.textContent = 'Save all'
+  }, 1500)
```
That's too short for me
```suggestion
  }, 3000)
```

angelikatyborska commented on popup/index.html:
```diff
@@ -1,81 +1,41 @@
+      <pre id="current-tab-title">(loading title)</pre>
+      <pre id="current-tab-slug">(loading slug)</pre>
+    </header>
+    <main>
```
This tag is not closed

angelikatyborska commented on options/index.html:
```diff
@@ -1,98 +1,75 @@
+            <button type="button" id="add-space">Add space</button>
+          </li>
+          <li class="submit">
+            <button type="submit" id="save-status">Save all</button>
+            <span role="alert" id="save-status"></span>
```
It's not valid HTML to have two elements with the same id. I assume you wanted to remove the span?

ironnysh replied:
Good catch

angelikatyborska commented on style.css:
```diff
@@ -0,0 +1,198 @@
+  justify-content: center;
+  align-items: baseline;
+  margin: 0;
+  background-color: var(--accent-dark);
```
I find the light and delicate buttons on an intense blue rectangle on a white page background uncomfortable to look at. There's too many contrast changes too close to one another. Can you make this background light in light mode and dark in dark mode, so there's not so much contrast?

<img width="649" alt="SCR-20250516-kihy" src="https://github.com/user-attachments/assets/82e34c68-0c9d-47f1-8bb8-47461fdc64e3" />

angelikatyborska commented on popup/index.html:
```diff
@@ -1,81 +1,41 @@
+
+  <body>
+    <header>
+      <pre id="current-tab-title">(loading title)</pre>
```
I don't really enjoy the addition of the page title. Many page titles and story slugs that I'm working with are very long, and are going to take up a lot of space. The pages I work on also often contain repetitive parts in their titles that aren't part of the story title, like the `| Storyblok`. It's not that useful to see there. Could you remove the title?

<img width="813" alt="SCR-20250516-kkhm" src="https://github.com/user-attachments/assets/c5295552-380c-4d6e-9192-1de7dc269e64" />

angelikatyborska commented on popup/index.js:
```diff
@@ -1,70 +1,95 @@
-          linkWrapper.textContent = 'Storyblok story not found'
-        }
-      }).catch(e => console.error('fetch error', e))
+                link.textContent = `Edit ${storyTitle}`;
```
I would really prefer the link text to remain "open story on Storyblok". I don't like the story title in there for those reasons:
- I want to rely on my muscle memory so I can click the link fast and without thinking. For this to be possible, the link needs to be the same size and in the same position every time, regardless of the story.
- The title can be really long sometimes
- The page title is not necessairly equal to the story title, like in the example below (The story title is just "Not Found", the page title is "Not Found | Storyblok")

<img width="431" alt="SCR-20250516-kmxv" src="https://github.com/user-attachments/assets/dcffb18a-51c5-4a10-a277-a376c83229e9" />

ironnysh replied:
All good points, thanks for explaining :-)

angelikatyborska commented on style.css:
```diff
@@ -0,0 +1,198 @@
+  }
+  &:active {
+    transform: scale(99%);
+  }
```
Does this do anything? I don't see any difference when pressing buttons

angelikatyborska commented on style.css:
```diff
@@ -0,0 +1,198 @@
+    background-color: var(--text);
+    color: canvas;
+    border: 0.2px solid transparent;
+  }
```
I can't really see any changes to the button styles on hover:

![2025-05-16 11 44 15](https://github.com/user-attachments/assets/a031438f-7bd1-4b01-9324-32ac829dc969)

angelikatyborska commented on style.css:
```diff
@@ -0,0 +1,198 @@
+  --accent-dark: #184db5;
+  --text: light-dark(var(--dark), var(--light));
+  --surface: light-dark(var(--light), var(--dark));
+  --alternate-accent: light-dark(var(--accent), var(--accent-dark));
```
Something is not right with the light / dark mode switching. This is how the settings page looks like in Firefox with a dark system theme:

<img width="1313" alt="SCR-20250516-kpzf" src="https://github.com/user-attachments/assets/1874baaa-765f-481b-9ca6-8284bd4a5356" />

angelikatyborska commented on popup/index.html:
```diff
@@ -1,81 +1,41 @@
+      </thead>
+      <tbody id="spaces"></tbody>
+    </table>
+    <p>Add Storyblok spaces in the <a href="../options/index.html" target="_blank">Extension Options</a>.</p>
```
I didn't know linking to extension options like that is going to work. Nice!

angelikatyborska commented on style.css:
```diff
@@ -0,0 +1,198 @@
+
+tbody > tr:nth-of-type(even) {
+  background-color: hsl(from var(--surface) h s calc(l - 10));
+}
```
Does this style work? For me, all table rows have the same background color

<img width="548" alt="SCR-20250516-kuwa" src="https://github.com/user-attachments/assets/1cd18946-12a8-459b-99ea-e783745b94f1" />

angelikatyborska commented on style.css:
```diff
@@ -0,0 +1,198 @@
+}
+
+.space-id {
+  font-variant-numeric: tabular-nums;
```
Cool style, I didn't know about this before

ironnysh replied:
Yeah, there are [dozens](https://developer.mozilla.org/en-US/docs/Web/CSS/font-variant) of these, depending on the font

angelikatyborska reviewed (changes requested):
Thank you for the contribution 🙏 I did initial testing in Firefox and left a few comments with my thoughts. Later, I would also do some testing in Chrome, but let's see first if we can resolve my comments.

Not gonna lie, I think we have very different taste in web design 😅 There are a few changes that I don't really enjoy, like the differing font stretch and usage of page titles. But there are also some very good changes here, like better documentation, a link to settings from the the popup, and a better layout for the settings page. Let's see if we can reach some compromises 🙂.

PS: I won't be offended if you choose not to spend any more time on this PR and ignore it. We all have any so much time in a day.

ironnysh committed:
Revert change in favor of a local .gitignore

ironnysh committed:
Update date

ironnysh committed:
Minor copy changes

ironnysh committed:
Revert change
(https://github.com/angelikatyborska/open-on-storyblok-extension/pull/1#discussion_r2092602884)

ironnysh committed:
Update screenshot.avif

ironnysh committed:
Remove link to the extension's repo

ironnysh committed:
Get and display extension name and version

ironnysh committed:
Remove page title, a11y fix, get link to settings page

ironnysh committed:
Remove page title references, add autofocus to story link, get link to
settings page

ironnysh committed:
Style updates and fixes

ironnysh commented:
Thanks for your thoughtful review 🙏 I really appreciate the reasoning behind everything. And I immediately adopted the local `.gitignore` tip! 

I updated and fixed a bunch of things (though I might have gotten carried away with the color palette, so feel free to go back to hex 😉). I hope this design works better :-))

I'll do a final round of tests on Chrome, Safari, and Firefox, and push the update soon.

ironnysh committed:
Add form pattern validation, improve a11y
